clients/drcachesim/simulator/simulator.cpp

```diff
@@ -87,6 +88,21 @@ simulator_t::init_knobs(unsigned int num_cores, uint64_t skip_refs, uint64_t war
     }
 }
 
+void
+simulator_t::create_v2p_from_file(std::string v2p_file_path)
```

Comment:
Another consideration is to allow i/o environments where regular file i/o is not possible: for that we try to pass in an `istream *` or we make things virtual so they can be overridden.  This would have to include `create_v2p_info_from_file` which is the one actually opening the file.  I would suggest testing an internal tlb simulator build on internal file systems and ensure you can override or otherwise get the right data (it might come from a protobuf instead of a text file; would it have a separate reader?).

Reply:
Using `std::ifstream` like for the config file of cache_simulator.
It would need a separate reader if it was coming from a protobuf.